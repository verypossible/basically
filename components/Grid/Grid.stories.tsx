import * as React from 'react'

import { text, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import info from '../../utils/storybook-info'
import Card from '../Card'
import Grid from './index'

const withInfo = info(`
### Notes

The notes need to be updated for the new component.

### Customization

\`\`\`
import { Card, Grid } from 'basically'
import styled from 'styled-components'

const MyGrid = styled(Grid)\`
  background: blue;
  color: white;
\`;

<MyGrid>
  <Card>I'm Customized</Card>
  <Card>Look at Me</Card>
</MyGrid>
\`\`\`
`)

const stories = storiesOf('Components/Grid', module)
stories.addDecorator(withKnobs)

stories.add('Documentation', withInfo(() => {
  return (
    <Grid>
      <Card>{text('card 1', 'Hello')}</Card>
      <Card>{text('card 2', 'There')}</Card>
      <Card>{text('card 3', 'Friend')}</Card>
    </Grid>
  )
}))
