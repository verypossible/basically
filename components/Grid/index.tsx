import * as React from 'react'
import styled from 'styled-components'

interface IGrid {
  className?: string
  children: JSX.Element[] | JSX.Element | string
}

const UnStyledGrid = ({ children, ...props }: IGrid) => {
  return <div {...props}>{children}</div>
}

const StyledGrid = styled(UnStyledGrid)`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(280px, 1fr));
  gap: 1rem 1rem;
`

export const Grid = (props: IGrid) => <StyledGrid {...props} />

export default Grid
