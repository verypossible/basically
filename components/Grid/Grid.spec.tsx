import * as React from 'react'

import { shallow } from 'enzyme'
import 'jest-styled-components'

import Grid from './index'

describe('Grid', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Grid>Hi There</Grid>)
    expect(tree).toMatchSnapshot()
  })
})
