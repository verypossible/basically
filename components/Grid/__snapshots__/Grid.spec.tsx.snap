// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid matches the snapshot 1`] = `
<Styled(UnStyledGrid)>
  Hi There
</Styled(UnStyledGrid)>
`;
